### ISSUE TITLE
Enhance RKE2 Cluster Templates docs

### ISSUE THREAD

**Issue by btat** [rancher/rancher-docs#52]
The current page (https://rancher.com/docs/rancher/v2.6/en/admin-settings/cluster-templates/) is a good starting point, but it does not cover the same level of details as the RKE1 template page (https://rancher.com/docs/rancher/v2.6/en/admin-settings/rke-templates/).

Review the RKE1 version (and its subpages) and add the corresponding information for RKE2.

**Comment by staedter** [comment]
Yes, please update the current docs and the official cluster template examples repo. We tried using them implementing a IaC-first approach using rancher on top of harvester.

And although I think most of what we want from Rancher to accomplish that, is already technically feasible. Trying implementing it ourself from scratch using the existing docs is/was a PITA.

For example there are multiple errors in the docs.
Case in point: how to label and taint the new nodes. The helpful documentation is buried somewhere under server configuration. https://ranchermanager.docs.rancher.com/reference-guides/cluster-configuration/rancher-server-configuration/rke2-cluster-configuration

The problem with the examples is, that they put the nodepool labels inside the harvesterconfig.yaml for the nodepool
![image](https://github.com/rancher/rancher-docs/assets/46670623/c6c31c95-e5c0-4344-8090-29608849a83d)

 but they need to be under the machinepool config in the cluster.yaml
![image](https://github.com/rancher/rancher-docs/assets/46670623/3f2baadc-2a40-4f2f-b46a-d8e75da7a2d4)

We found this out only after a lot of trial an error. Also the config from the harvesterconfig seems problematic. Because the `common:` field is something that throws errors when I tried to apply it with helm directly.

Furthermore the CNI is hardcoded in the examples to calico and the list goes on and on.
![image](https://github.com/rancher/rancher-docs/assets/46670623/9c38c17f-7e16-430b-8a1e-8732fe4b1492)

At least there was the Youtube Master Class video with @bashofmann and his example repo at https://github.com/bashofmann/rancher-cluster-templates/ which made some points clearer, but this were just his demo projects and the also do not cover our harvester cloudprovider/nodedriver use case.

Some open questions I have:

- Why are the rancher-node hostnames different from the harvester-machine names? And how can I fix this using the Templates?
- Why is the builtin ssh-shell not working?
- How do I inject a specific ssh key for the builtin user?
- How do I switch the builtin ssh user (because while using the templates does switch the ssh user in the harvesterconfig, the downloaded ssh key from rancher only works with the default ubuntu user) 
- (How) Can I use cloudconfig templates from inside harvester (which is possible when creating a harvester cluster from the UI) inside the cluster templates, and will this work while also injecting the needed custom_install.sh script that is apparently necessary for bootstrapping the RKE2 cluster?

I have alot more of similar questions, some of which I unsuccessfully tried to get answers to in the rancher user slack and so I am hoping that some of the answers may find its way into the updated docs. If someone wants to get in touch I could also provide even more examples where the docs were misleading/unhelpful, please feel free to reach out.

Best regards 
Chris
